R/nm-tables.R

```diff
@@ -84,14 +84,12 @@ nm_tables <- function(
 #' @export
 nm_table_files <- function(.mod, .check_exists = TRUE) {
   .p <- get_model_path(.mod)
-  .l <- parse_ctl_to_list(.p)
+  .l <- nmrec::read_ctl(.p)
   out_dir <- get_output_dir(.mod, .check_exists = .check_exists)
 
   # get file names from table statements and construct paths
-  .f <- .l[grep("^TAB", names(.l))] %>%
-    map_chr(~paste(.x, collapse = " ")) %>%
-    str_extract("\\bFILE\\s*=\\s*([^ ]+)") %>%
-    str_replace("\\bFILE\\s*=\\s*", "") %>%
+  .f <- nmrec::select_records(.l, "table") %>%
+    purrr::map_chr(~nmrec::get_record_option(.x, "file")$value) %>%
```

Comment:
A table record doesn't necessarily have a `FILE` option, in which case `NULL` will be returned by this function. `map_chr` will signal an error in that case. To stick with `map_chr`, you could tack on an `... %||% NA_character_` at the end of the function body and then filter NAs out downstream, but it's probably cleaner to instead do

```R
purrr::map(...) %>%
  unlist()
```

Reply:
fixed